controllers/managedocs_controller_test.go

```diff
@@ -1139,8 +1170,8 @@ var _ = Describe("ManagedOCS controller", func() {
 			})
 			When("a Grafana datasources secret exists in the openshift-monitoring namespace", func() {
 				It("should create k8sMetricsServiceMonitorAuthSecret in primary namespace", func() {
-					grafanaSecret := grafanaDatasourceSecretTemplate.DeepCopy()
-					Expect(k8sClient.Create(ctx, grafanaSecret)).Should(Succeed())
+					secret := grafanaDatasourceSecretTemplate.DeepCopy()
+					Expect(k8sClient.Get(ctx, utils.GetResourceKey(secret), secret)).Should(Succeed())
```

Comment:
if you assume the secret is already created you should not test for it, 
Expect is used to validate test results, not assumptions